### PR TITLE
Refactoring hashtables, hashnode, and repeatedWord

### PR DESCRIPTION
**Pull request by sajafisher29**

### CODE REVIEW

File: 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java [hunk 1]
```diff
@@ -91,42 +87,42 @@ public boolean add(String key, String value) {
     // **************** Get Method ***************
 
     // Returns value for a key
-    public String get(String key) {
+    public Value get(Key key) {
 
         int index = hash(key);
+        if (bucketArray[hashCode()] == null) {
```
Comment (ncarignan) [inline 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java]: you do not use your index on line 93, replace hashCode() with index

File: 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java [hunk 2]
```diff
@@ -91,42 +87,42 @@ public boolean add(String key, String value) {
...
 
     // Assess whether the specified key is stored in the table.
 
-    public boolean contains(String key){
+    public boolean contains(Key key){
         int index = hash(key);
-        LinkedList<HashNode> indexsLinkedList = this.bucketArray[index];
 
-        boolean result = false;
+        if (bucketArray[hashCode()] == null) {
```
Comment (ncarignan) [inline 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java]: replace hashCode() with index

File: 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java (hunk 1 above)
Comment (sajafisher29) [reply 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java]: Thank you! Refactored.

File: 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java (hunk 2 above)
Comment (sajafisher29) [reply 401codechallenges/src/main/java/code401challenges/hashtable/Hashtable.java]: Refactored.

**Pull request merged**